From d1540192d87f2f4fd20ce9b778cfea6f2b985796 Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Fri, 8 Jun 2012 16:44:03 +0200
Subject: [PATCH 07/78] wifi: add missing wake up queues when suspend fails

BZ: 20670

When wifi driver fails to suspend, queues are never woken up.
This can end up in wifi not able to connect.
Fix this by explicitely waking up the queues when driver suspend
fails.

Change-Id: I962f22280b9f31475ea724ea9564591e62ad628f
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Reviewed-on: http://android.intel.com:8080/52165
Reviewed-by: Regairaz, Julie <julie.regairaz@intel.com>
Reviewed-by: Trivelly, Jean <jean.trivelly@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/59013
Reviewed-on: http://android.intel.com:8080/77964
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/net/mac80211/pm.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/wl12xx-compat/net/mac80211/pm.c b/wl12xx-compat/net/mac80211/pm.c
index 596efaf..e6ee9ae 100644
--- a/wl12xx-compat/net/mac80211/pm.c
+++ b/wl12xx-compat/net/mac80211/pm.c
@@ -77,6 +77,8 @@ int __ieee80211_suspend(struct ieee80211_hw *hw, struct cfg80211_wowlan *wowlan)
 		int err = drv_suspend(local, wowlan);
 		if (err < 0) {
 			local->quiescing = false;
+			ieee80211_wake_queues_by_reason(hw,
+					IEEE80211_QUEUE_STOP_REASON_SUSPEND);
 			return err;
 		} else if (err > 0) {
 			WARN_ON(err != 1);
-- 
1.7.9

